src/mlpack/methods/ann/layer/threshold_fn.hpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file methods/ann/layer/threshold_fn.hpp
+ * @author Shubham Agrawal
+ *
+ * Definition of the Threshold activation function
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_THRESHOLD_HPP
+#define MLPACK_METHODS_ANN_LAYER_THRESHOLD_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Threshold activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     x & : x > threshold \\
+ *     value & : x \le threshold
+ *   \end{array}
+ * \right.
+ * @f}
+ *
```

Comment:
Is there any reference to this? Would you add the article that proposed this activation function as a reference?

Reply:
I have only PyTorch docs as a reference.

Reply:
@shrit

Reply:
I don't think we will find a paper for the activation function, it#s super simple, so from my side this looks fine.